packages/gatsby-transformer-react-docgen/src/extend-node-type.js

```diff
@@ -0,0 +1,73 @@
+const {
+  GraphQLBoolean,
+  GraphQLObjectType,
+  GraphQLList,
+  GraphQLString,
+} = require(`graphql`)
+const GraphQLJSON = require(`graphql-type-json`)
+
+function resolve(field) {
+  return obj => JSON.parse(obj.content)[field]
+}
+
+const PropDefaultValue = new GraphQLObjectType({
+  name: `PropDefaultValue`,
+  fields: () => ({
+    value: { type: GraphQLJSON },
+    computed: { type: GraphQLBoolean },
+  }),
+})
+
+const Method = new GraphQLObjectType({
+  name: `ComponentMethod`,
+  fields: () => ({
+    name: { type: GraphQLString },
+    docblock: { type: GraphQLString },
+    modifiers: { type: new GraphQLList(GraphQLJSON) },
+    params: {
+      type: new GraphQLList(
+        new GraphQLObjectType({
+          name: `ComponentMethodParams`,
+          fields: () => ({
+            name: { type: GraphQLString },
+            type: { type: GraphQLJSON },
+          }),
+        })
+      ),
+    },
+    returns: { type: new GraphQLList(GraphQLJSON) },
+  }),
+})
+
+function extendComponents() {
+  return {
+    doclets: {
+      type: GraphQLJSON,
+      resolve: resolve(`doclets`),
+    },
+    composes: {
+      type: new GraphQLList(GraphQLString),
+      resolve: resolve(`composes`),
```

Comment:
Originally I couldn't because of that bug in the input generation... Also I was worried that they wouldn't always be there depending on the sites components. Other than that tho I was just being explicit since I know the shape. 

Honestly for the simpler ones here there isn't any good reason it's this way :P